### PR TITLE
Extracted xcoderunner (raw xcode, xcpretty) behind an inteface

### PR DESCRIPTION
**Pull request by lpusok**
### Checklist
- [x] I've read and followed the [Contribution Guidelines](https://github.com/bitrise-steplib/.github/blob/main/CONTRIBUTING.md)
- [x] `step.yml` and `README.md` is updated with the changes (if needed)

### Version

Requires a *PATCH* [version update](https://semver.org/)

### Context

Created an interface for xcode command runner (just xcodebuild, xcpretty). This makes easier to add new ones.

### Changes

- Created new *xcodecommand* package, this exports an interface for: running _xcodebuild_ in different ways (using _xcpretty_ or not), and installing runner tools (like _xcpretty_).
  - Moved all xcpretty related code here (from dep: _go-xcode/v2/xcpretty_  and _xcpretty_ package)
- Extracted _XcodeTestConfigParser_ from _XcodeTestRunner_. This was needed so _XcodeTestRunner_ can be created after the config is parsed; so can inject different implementations based on the log formatter (selected in inputs).
- Removed now unused go-xcode/v2/xcpretty dep

### CODE REVIEW

File: xcodecommand/xcpretty.go [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+		}, err
+	}
+	if err := prettyCmd.Start(); err != nil {
+		return Output{
+			RawOut:   buildOutBuffer.Bytes(),
+			ExitCode: 1,
+		}, err
+	}
+
+	defer func() {
```
Comment (ofalvai) [inline xcodecommand/xcpretty.go:62]: This should be higher in the function. At this point there are two return statements above this.

File: xcodecommand/xcpretty_helper.go [hunk 2]
```diff
@@ -0,0 +1,39 @@
+package xcodecommand
+
+import (
+	"github.com/bitrise-io/go-steputils/v2/ruby"
+	command "github.com/bitrise-io/go-utils/v2/command"
+	version "github.com/hashicorp/go-version"
+)
+
+type xcprettyManager interface {
```
Comment (ofalvai) [inline xcodecommand/xcpretty_helper.go:9]: Wouldn't it make sense to merge this whole helper file into `xcpretty_installer.go`? These methods mostly do delegation to other methods.

File: main.go [hunk 3]
```diff
@@ -58,24 +64,47 @@ func run() int {
...
+	if logFormatter == xcodebuild.XcprettyTool {
+		commandLocator := env.NewCommandLocator()
+		rubyComamndFactory, err := ruby.NewCommandFactory(commandFactory, commandLocator)
+		if err != nil {
+			return step.XcodeTestRunner{}, fmt.Errorf("failed to install xcpretty: %s", err)
+		}
+		rubyEnv := ruby.NewEnvironment(rubyComamndFactory, commandLocator, logger)
+
+		xcodeCommandRunner = xcodecommand.NewXcprettyCommandRunner(logger, commandFactory)
+		xcodeRunnerDepInstaller = xcodecommand.NewXcprettyDependencyManager(logger, commandFactory, rubyComamndFactory, rubyEnv)
```
Comment (ofalvai) [inline main.go:105]: If we now have all this new code for installing and running xcpretty, I feel that this setup code shouldn't be in `main.go`.

File: main.go [hunk 4]
```diff
@@ -58,24 +64,47 @@ func run() int {
...
-	deviceFinder := destination.NewDeviceFinder(logger, commandFactory)
 	simulatorManager := simulator.NewManager(logger, commandFactory)
 	swiftCache := cache.NewSwiftPackageCache()
 	testAddonExporter := testaddon.NewExporter(testaddon.NewTestAddon(logger))
 	stepenvRepository := stepenv.NewRepository(envRepository)
 	outputExporter := output.NewExporter(stepenvRepository, logger, testAddonExporter)
 	utils := step.NewUtils(logger)
 
-	return step.NewXcodeTestRunner(inputParser, logger, xcprettyInstaller, xcodebuilder, deviceFinder, simulatorManager, swiftCache, outputExporter, pathModifier, pathProvider, utils)
+	xcodeCommandRunner := xcodecommand.NewRawCommandRunner(logger, commandFactory)
```
Comment (ofalvai) [inline main.go:94]: nit: I find it a bit confusing that this value gets reassigned if `logFormatter` is `xcpretty`, a switch statement would make it easier to read.

Comment (lpusok) [reply main.go:94]: Will fix in follow-up PR.

File: main.go (hunk 3 above)
Comment (lpusok) [reply main.go:105]: Technically it is not installing anything at this point. Should we move it to a new file?

File: xcodecommand/xcpretty.go (hunk 1 above)
Comment (lpusok) [reply xcodecommand/xcpretty.go:68]: Moved pipe close after declaration. Wait is only needs to be called if Start() is successful, left that in place.

File: xcodecommand/xcpretty_helper.go (hunk 2 above)
Comment (lpusok) [reply xcodecommand/xcpretty_helper.go]: Yes, moved it. Only split in a new file so that the diff engine picks it as a moved file but it did not.

**Review by ofalvai — DISMISSED** [review]

**Review by ofalvai — APPROVED** [review]

**Pull request merged**